docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+     - unsigned integer
+     - 8
+     - little-endian
+   * - ``>u2``
+     - unsigned integer
+     - 2
+     - big-endian
+   * - ``>u4``
+     - unsigned integer
+     - 4
+     - big-endian
+   * - ``>u8``
+     - unsigned integer
+     - 8
+     - big-endian
+   * - ``<f2``
+     - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
+     - 2
+     - little-endian
+   * - ``<f4``
+     - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
+     - 4
+     - little-endian
+   * - ``<f8``
+     - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
+     - 8
+     - little-endian
+   * - ``>f2``
+     - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
+     - 2
+     - big-endian
+   * - ``>f4``
+     - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
+     - 4
+     - big-endian
+   * - ``>f8``
+     - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
+     - 8
+     - big-endian
+   * - ``r*`` (Optional)
+     - raw bits,  use for extension type fallbacks
+     - variable, given by ``*``, is limited to be a multiple of 8.
+     - N/A
+
+
+Floating point types correspond to basic binary interchange formats as
+defined by IEEE 754-2008.
+
+Additionally to these base types, an implementation should also handle the
+raw/opaque pass-through type designated by the lower-case letter ``r`` followed
+by the number of bits, multiple of 8. For example, ``r8``, ``r16``, and ``r24``
+should be understood as fall-back types of respectively 1, 2, and 3 byte length.
+
+Zarr v3 is limited to type sizes that are a multiple of 8 bits but may support
+other type sizes in later versions of this specification.
+
+
+.. note::
+
+    We are explicitly looking for more feedback and prototypes of code using the ``r*``,
+    raw bits, for various endianness and whether the spec could be made clearer.
+
+.. note::
+
+    Currently only fixed size elements are supported as a core data type.
+    There are many request for variable length element encoding. There are many
+    ways to encode variable length and we want to keep flexibility. While we seem
+    to agree that for random access the most likely contender is to have two
+    arrays, one with the actual variable length data and one with fixed size
+    (pointer + length) to the variable size data, we do not want to commit to such
+    a structure.
+
+
+Chunk grids
+===========
+
+A chunk grid defines a set of chunks which contain the elements of an
+array. The chunks of a grid form a tessellation of the array space,
+which is a space defined by the dimensionality and shape of the
+array. This means that every element of the array is a member of one
+chunk, and there are no gaps or overlaps between chunks.
+
+In general there are different possible types of grids. The core
+specification defines the regular grid type, where all chunks are
+hyperrectangles of the same shape. Extensions may define other grid
```

Comment:
I think it would be good to be very explicit about the size of border chunks here (e.g. `shape=(30, 30)`, `chunks=(16, 16)`). As far as I understand the current spec the border chunks would have size `(16, 16)`, where the rows / columns not part of the data space don't hold valid data. (It probably makes most sense to add this as a `note` to the "Regular grids" section.)

Reply:
Added this to #171, see commit f889dad. Marking as resolved for now.